components/tabs/tabs-label.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useMemo, useEffect, useRef, useState, CSSProperties } from 'react'
+import { TabStatus, TabVarient } from 'components/utils/prop-types'
+export type LabelCpt = React.FC<{
+  label: string
+  status: TabStatus
+  varient: TabVarient
+  colors: any
+}>
+
+type CSS = { [key in keyof CSSProperties]: CSSProperties[key] }
+const Label: LabelCpt = ({ label, varient, status, colors }) => {
+  const [width, ref] = useFixedWidth<HTMLDivElement>()
+  // console.log('width', width)
+  const extra = useExtraStyle(varient, status)
+  return (
+    <div style={{ ...colors, ...extra, width: width ? width : '' }} ref={ref} className="label">
```

Comment:
Why combine inline style with styled-jsx here?

Reply:
`...colors` and `...extra` should be inline because they are not declared in the component

Reply:
Why use the `useDOMDimension` to control the width? Is it possible to for user to control this width anymore?

Reply:
Shouldn't the width be controlled by css?